web/src/main/java/org/akaza/openclinica/control/submit/InitialDataEntryServlet.java

```diff
@@ -43,6 +46,35 @@
     protected final Logger logger = LoggerFactory.getLogger(getClass().getName());
     Locale locale;
 
+    /**
+     * 
+     * @param request
+     * @throws InsufficientPermissionException
+     */
+    private void checkUpdateDataPermission(HttpServletRequest request) throws InsufficientPermissionException {
+		  Boolean auth = true;
+		  UserAccountBean userBean =(UserAccountBean) request.getSession().getAttribute(USER_BEAN_NAME);
+		  ArrayList userRoles = userBean.getRoles();
+		  String submitted =(String) request.getParameter("submitted");
+		  String checkInputs = (String)request.getParameter("checkInputs");
+		  
+		  if(submitted !=null && submitted.equals("1") && checkInputs!=null && checkInputs.equals("1")) {
+			  for (int i = 0; i < userRoles.size(); i++) {
+		            StudyUserRoleBean studyRole = (StudyUserRoleBean) userRoles.get(i);
+
+					if(studyRole.getRole().equals(Role.MONITOR))
+					{
+						auth = false;						
```

Comment:
This will prevent users with Monitor access to any of the studies from entering data. We should check specific study's user role.

Reply:
done